File: src/main/resources/static/js/validate/schema/dctnInfo.js
const DCTN_INFO_VALID = [
    {
        key: "abbrnm",
        type: "input",
        name: "약어명",
        isInsert: true,
        valid: true,
        notNull: true,
        size: {
            min: 1,
            max: 50
        }
    },
    {
        key: "word_nm",
        type: "input",
        name: "단어 명",
        isInsert: true,
        valid: true,
        notNull: true,
        size: {
            min: 1,
            max: 300
        }
    },
    {
        key: "engnm",
        type: "input",
        name: "영문명",
        isInsert: true,
        valid: true,
        notNull: true,
        size: {
            min: 1,
            max: 300
        }
    },
    {
        key: "word_expln",
        type: "textarea",
        name: "단어 설명",
        isInsert: true,
        valid: true,
        notNull: true,
        size: {
            min: 1,
            max: 4000
        }
    },
    {
        key: "clsfct",
        type: "select",
        selectOptions: [
            {text: "국방데이터", value: "ND"},
            {text: "공공데이터", value: "PBL"},
            {text: "유저데이터", value: "USER"}
        ],
        name: "분류 코드",
        isInsert: true,
        valid: true,
        notNull: true,
        size: {
            min: 1,
            max: 15
        }
    },
    {
        key: "wrtr",
        type: "input",
        name: "작성자",
        isInsert: true,
        valid: true,
        notNull: true,
        size: {
            min: 1,
            max: 50
        }
    },
    {
        key: "crt_dttm",
        type: "span",
        name: "생성 일시",
        isInsert: false,
        valid: false,
        notNull: false,
        size: {
            min: 0,
            max: 0
        }
    },
    {
        key: "mdfr",
        type: "input",
        name: "수정자",
        isInsert: false,
        valid: true,
        notNull: false,
        size: {
            min: 1,
            max: 50
        }
    },
    {
        key: "mdfr_rsn",
        type: "input",
        name: "수정 사유",
        isInsert: false,
        valid: true,
        notNull: false,
        size: {
            min: 1,
            max: 500
        }
    },
    {
        key: "mdfc_dttm",
        name: "수정 일시",
        type: "span",
        isInsert: false,
        valid: false,
        notNull: false,
        size: {
            min: 0,
            max: 0
        }

    }
]

File: src/main/java/egovframework/cbm/web/dctn/model/entity/DctnInfo.java
package egovframework.cbm.web.dctn.model.entity;

import egovframework.cbm.web.common.model.entity.CmncdInfo;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.*;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 데이터 사전 정보 엔티티
 *
 * @author  이상민
 * @since   2024.07.08 12:00
 */
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
@Table(name = "DCTN_INFO")
public class DctnInfo implements Serializable {

    @Serial
    private static final long serialVersionUID = -6877217338108617548L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "DCTN_INFO_SEQ", nullable = false, updatable = false)
    private int dctnInfoSeq;

    @Column(name = "ABBRNM", nullable = false, length = 50)
    private String abbrnm;

    @Column(name = "WORD_NM", nullable = false, length = 300)
    private String wordNm;

    @Column(name = "WORD_EXPLN", nullable = false, length = 4000)
    private String wordExpln;

    @Column(name = "ENGNM", nullable = false, length = 300)
    private String engnm;

    @Column(name = "CLSFCT", nullable = false, length = 15)
    private String clsfct;

    @Column(name = "WRTR", nullable = false, updatable = false, length = 50)
    private String wrtr;

    @CreationTimestamp
    @Column(name = "CRT_DTTM", nullable = false, updatable = false)
    private LocalDateTime crtDttm;

    @UpdateTimestamp
    @Column(name = "MDFC_DTTM", insertable = false)
    private LocalDateTime mdfcDttm;

    @Column(name = "MDFR", insertable = false, length = 50)
    private String mdfr;

    @Column(name = "MDFR_RSN", insertable = false, length = 500)
    private String mdfrRsn;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CLSFCT", referencedColumnName = "CD", updatable = false, insertable = false)
    private CmncdInfo cmncdInfo;

    @OneToOne(mappedBy = "dctnInfo")
    private ApprvPtcls apprvPtcls;

}
